packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -400,31 +407,48 @@ export function renderWithHooks<Props, SecondArg>(
   // didScheduleRenderPhaseUpdate = false;
   // localIdCounter = 0;
 
-  // TODO Warn if no hooks are used at all during mount, then some are used during update.
-  // Currently we will identify the update render as a mount because memoizedState === null.
-  // This is tricky because it's valid for certain types of components (e.g. React.lazy)
-
-  // Using memoizedState to differentiate between mount/update only works if at least one stateful hook is used.
-  // Non-stateful hooks (e.g. context) don't get added to memoizedState,
-  // so memoizedState would be null during updates and mounts.
-  if (__DEV__) {
-    if (current !== null && current.memoizedState !== null) {
```

Comment:
Seems excessive to fork this entire block.

Instead I would change this line to:

```js
if (current !== null && (current.mode & ConcurrentMode || current.memoizedState !== null)) {
```

Reply:
Done